pac4j-saml/src/main/java/org/pac4j/saml/client/SAML2ClientConfiguration.java

```diff
@@ -59,25 +67,56 @@ public SAML2ClientConfiguration() {}
     private boolean wantsAssertionsSigned = true;
 
-    public SAML2ClientConfiguration(final String keystorePath, final String keystorePassword,
-                                    final String privateKeyPassword, final String identityProviderMetadataPath) {
-        this(keystorePath, keystorePassword, privateKeyPassword, identityProviderMetadataPath, null, null);
-    }
+	private String keyStoreAlias;
+
+	private String keyStoreType;
+
+	public SAML2ClientConfiguration(final KeyStore keystore, 
+			                        final String keyStoreAlias, 
+			                        final String keyStoreType,
+			                        final String privateKeyPassword, 
+			                        final Resource identityProviderMetadataResource) {
+		this(keystore, keyStoreAlias, keyStoreType, null, null, null, privateKeyPassword,
+				identityProviderMetadataResource, null, null, null);
+	}
+
+	public SAML2ClientConfiguration(final Resource keystoreResource, final String keyStoreAlias,
+			final String keyStoreType, final String keystorePassword, final String privateKeyPassword,
+			final Resource identityProviderMetadataResource) {
+		this(null, keyStoreAlias, keyStoreType, keystoreResource, null, keystorePassword, privateKeyPassword,
+				identityProviderMetadataResource, null, null, null);
+	}
 
     public SAML2ClientConfiguration(final String keystorePath, final String keystorePassword,
-                                    final String privateKeyPassword, final String identityProviderMetadataPath,
-                                    final String identityProviderEntityId, final String serviceProviderEntityId) {
-        this.keystorePath = keystorePath;
+                                    final String privateKeyPassword, final String identityProviderMetadataPath) {
+        this(null, null, null, null, keystorePath, keystorePassword, privateKeyPassword, null,
+                identityProviderMetadataPath, null, null);
+    }
+
+	private SAML2ClientConfiguration(final KeyStore keyStore, final String keyStoreAlias, final String keyStoreType,
+			                         final Resource keystoreResource, final String keystorePath, final String keystorePassword,
+                                     final String privateKeyPassword, final Resource identityProviderMetadataResource,
+			                         final String identityProviderMetadataPath, 
+                                     final String identityProviderEntityId, final String serviceProviderEntityId) {
+		this.keyStore = keyStore;
+        this.keyStoreAlias = keyStoreAlias;
+        this.keyStoreType = keyStoreType;
+        this.keystoreResource = keystoreResource;
+        if (this.keystoreResource == null) {
+        	this.keystoreResource = CommonHelper.getResource(keystorePath);
+        }
         this.keystorePassword = keystorePassword;
         this.privateKeyPassword = privateKeyPassword;
+		this.identityProviderMetadataResource = identityProviderMetadataResource;
```

Comment:
Cannot we do like for the `keystorePath` and `keystoreResource` regarding the `identityProviderMetadataPath` and `identityProviderMetadataResource`?